### PR TITLE
Modular bounds

### PR DESCRIPTION
**Pull request by NotJeven**
Implementation of issue [#102](https://github.com/Moesh/Calamity/issues/102)

Using barrier block "data layer" at y=74 for the bounds, and y=73 for no build walls, this will find and replace moving_piston blocks right under the barrier blocks that are acting to limit the map height. The map height barrier blocks can be moved anywhere. This removed all the hard coded coordinates for stopping blocks from being placed on the map height barrier blocks. This is the same system as the moving_piston walls applied over the map height. The trade off is that for this to make sense I had to place moving_pistons under the map height barrier blocks (I did it for both the region/ and generated/ files).

This is a proof of concept but works as intended. I would want to rewrite the search algorithms for finding where to place a moving_piston. Brief QA also tells me that I need to run the search algorithm for the entire map height barrier blocks and not just in the in bounds area. The map height barrier blocks should probably be raised 1 block to allow for moving_pistons over top the mine. Something will have to be done to make the moving_piston wall height adjustable too. The problem arises in `calamity:build_protection/fix_wall`. Some standards also need to be established for where the data layers should be but I think I will raise this in another issue.

TLDR: Added the same build protection system the walls use to the map height barrier blocks. This removed having to use hard coded coordinates to prevent placing blocks on the map height barrier blocks.

### CODE REVIEW

**Review by Vilder50 — COMMENTED** [review]
The old "Can't place blocks under barrier blocks system" was in place to stop people from placing blocks under the barrier blocks outside the arena.
The barrier blocks outside the arena can be removed now though since they are no longer used (They were used for stopping players from blowing up the old glass layer we had before we changed to barrier layers).

This pull request will again allow players to place blocks under the barriers outside the map, so those no longer needed barriers blocks outside should be removed (or something else should be done) before accepting this request.

Also this pull request doesn't add the moving piston layer to the arena "shattered", I don't know what the plans are for that arena so idk if those blocks should be added or not. Guess 
@Moesh would know.

A little bit of a nitpick: The newly added fill commands in `section_medium` and `handle_explosion` can be merged together with the wall checking commands. So instead of having a command filling layer y73 and then y74 it could be replaced with a single command filling from y73 to y74.

**Comment by Moesh** [comment]
Ignore Shattering for now--that level shouldn't even really be checked in. It was my example to show that I can already hot load multiple levels.

But...I am working on cool new Shattering things.

**Comment by NotJeven** [comment]
Yes, as noted, I am prototyping this to make sure that it does not collide with any other plans to change the arena height system.

I would want to rewrite the search algorithms. I would raise the calamity arena height barrier blocks by 1. I would remove the bounds limitation and place moving_pistons under all of the barriers.

Thinking about this raised a different issue for my though, if I have moving_piston blocks over the top of the in bounds area in the arena, why do I need the barriers at arena height? The moving pistons will block scaffolding from building. I prototyped a system using an arena height marker and the arena bounds that I think would make more sense.

That being said I am going to close this issue and work on that idea instead.
